app/javascript/app/pages/national-context/socioeconomic/population/population-selectors.js

```diff
@@ -0,0 +1,275 @@
+      config: {
+        axes: {
+          xBottom: { name: 'Years', unit: '', format: 'string' },
+          yLeft: { name: 'Number of people', unit: '', format: 'number' }
+        },
+        tooltip: {
+          y: {
+            label: 'People',
+            format: value => `${format(',.2r')(`${value * 1000}`)}`
+          },
+          x: { label: 'Year' },
+          indicator: selectedOptions[queryName] &&
+            selectedOptions[queryName].label
+        },
+        animation: false,
+        columns: {
+          x: [ { label: 'year', value: 'x' } ],
+          y: [
+            {
+              label: selectedOptions[queryName] &&
+                selectedOptions[queryName].label,
+              value: 'y'
+            }
+          ]
+        },
+        theme: { y: { stroke: '#2EC9DF', fill: '#2EC9DF' } },
+        yLabelFormat: getCustomYLabelFormat(unit)
+      },
+      dataOptions: getFirstChartFilter(queryName, selectedOptions),
+      dataSelected: getFirstChartFilter(queryName, selectedOptions)
+    };
+  }
+);
+
+const getPopProvinceBarChartData = createSelector(
+  [ getIndicators, getSelectedOptions ],
+  (indicators, selectedOptions) => {
+    if (!indicators || !selectedOptions) return null;
+
+    const INDICATOR_CODE = 'pop_total';
+    const queryName = 'popProvince';
+
+    const selectedIndicator = indicators &&
+      indicators.values &&
+      indicators.values.find(
+        ind =>
+          ind.location === selectedOptions[queryName].value &&
+            ind.indicator_code === INDICATOR_CODE
+      );
+
+    const indicator = indicators &&
+      indicators.indicators &&
+      indicators.indicators.find(ind => ind.code === INDICATOR_CODE);
+    const unit = indicator && indicator.unit;
+
+    const selectedData = [];
+    if (selectedIndicator) {
+      selectedIndicator.values.forEach(d => {
+        if (d.value && d.year) {
+          selectedData.push({ x: d.year, y: d.value });
+        }
+      });
+    }
+
+    return {
+      data: selectedData,
+      domain: { x: [ 'auto', 'auto' ], y: [ 0, 'auto' ] },
+      config: {
+        axes: {
+          xBottom: { name: 'Years', unit: '', format: 'string' },
+          yLeft: { name: 'Number of people', unit: '', format: 'number' }
+        },
+        tooltip: {
+          y: {
+            label: 'People',
+            format: value => `${format(',.2r')(`${value * 1000}`)}`
+          },
+          indicator: 'Population'
+        },
+        animation: false,
+        columns: {
+          x: [ { label: 'year', value: 'x' } ],
+          y: [ { label: 'Province population', value: 'y' } ]
+        },
+        theme: { y: { stroke: '#FC7E4B', fill: '#FC7E4B' } },
+        yLabelFormat: getCustomYLabelFormat(unit)
+      },
+      dataOptions: [ { label: 'Province population' } ],
+      dataSelected: [ { label: 'Province population' } ]
+    };
+  }
+);
+
+const getTranslatedContent = createSelector([ getSectionsContent ], data => {
```

Comment:
This is very Cool! 😎 Maybe we could even create a common selector for all components. Or maybe its too much